java/src/jmri/BlockManager.java

```diff
@@ -307,5 +312,50 @@ public String getBeanTypeHandled() {
         return blockList;
     }
 
+
+    private Instant lastTimeLayoutPowerOn; // the most recent time any power manager had a power ON event
+
+    /**
+     * Listen for changes to the power state from any power managers
+     * in use in order to track how long it's been since powet was applied
+     * to the layout. This information is used in {@link Block#goingActive()}
+     * when deciding whether to restore a block's last value.
+     *
+     * @param e
+     */
+
+    @Override
+    public void propertyChange(PropertyChangeEvent e) {
+        super.propertyChange(e);
+        if ("Power".equals(e.getPropertyName())) {
```

Comment:
You want to use the constant provided by PowerManager instead of the String literal.